src/core/StelCore.hpp

```diff
@@ -315,8 +316,13 @@ class StelCore : public QObject
 	const StelLocation& getCurrentLocation() const;
 	//! Get the UTC offset on the current location (in hours)
 	//! N.B. This is a rather costly operation. Re-use where possible!
+	//! In addition, this is now cached with 1-minute granularity.
 	double getUTCOffset(const double JD) const;
+private:
+	// Cache is over qHash(&UTCOffsetHashData).
+	static QCache<size_t, qint64> utcOffsetCache;
```

Comment:
Why static?

Reply:
If I make this a per-object variable, it breaks many const-qualified functions. If there will ever be more than one core, they would just share the results. Or do you see problems with this?

Reply:
> If I make this a per-object variable, it breaks many const-qualified functions.

This is exactly why `mutable` keyword was introduced: one of the uses for it is implementation of transparent caches.

> If there will ever be more than one core, they would just share the results.

Hmm, this doesn't seem like a likely situation, but in this way this might make sense, though there's some strangeness in the way you implement the cache that I can't tell whether it does.

Reply:
OK, I forgot about mutable.